java-strings/src/test/java/com/baeldung/string/StringFromPrimitiveArrayUnitTest.java

```diff
@@ -0,0 +1,96 @@
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Test;
+
+import java.nio.CharBuffer;
+import java.util.Arrays;
+import java.util.StringJoiner;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+public class StringFromPrimitiveArrayUnitTest {
+
+    private int[] intArray = {1, 2, 3, 4, 5, 6, 7, 8, 9};
+
+    private char[] charArray = {'a', 'b', 'c', 'd', 'e', 'f'};
+
+    private char separatorChar = '-';
+
+    private String separator = String.valueOf(separatorChar);
+
+    private String expectedIntString = "1-2-3-4-5-6-7-8-9";
+
+    private String expectedCharString = "a-b-c-d-e-f";
+
+    @Test
+    public void givenPrimitiveArray_whenJoinBySeparator_thenReturnsString_through_Java8CollectorsJoining() {
+        assertThat(Arrays.stream(intArray)
+          .mapToObj(String::valueOf)
+          .collect(Collectors.joining(separator)))
+          .isEqualTo(expectedIntString);
+
+        assertThat(CharBuffer.wrap(charArray).chars()
+          .mapToObj(intChar -> String.valueOf((char) intChar))
+          .collect(Collectors.joining(separator)))
+          .isEqualTo(expectedCharString);
+    }
+
+    @Test
+    public void givenPrimitiveArray_whenJoinBySeparator_thenReturnsString_through_Java8StringJoiner() {
+        StringJoiner intStringJoiner = new StringJoiner(separator);
+        Arrays.stream(intArray)
+          .mapToObj(String::valueOf)
+          .forEach(intStringJoiner::add);
+        assertThat(intStringJoiner.toString()).isEqualTo(expectedIntString);
+
+        StringJoiner charStringJoiner = new StringJoiner(separator);
+        CharBuffer.wrap(charArray).chars()
+          .mapToObj(intChar -> String.valueOf((char) intChar))
+          .forEach(charStringJoiner::add);
+        assertThat(charStringJoiner.toString()).isEqualTo(expectedCharString);
+    }
+
+    @Test
+    public void givenPrimitiveArray_whenJoinBySeparator_thenReturnsString_through_CommonsLang() {
+        assertThat(StringUtils.join(intArray, separatorChar)).isEqualTo(expectedIntString);
+        assertThat(StringUtils.join(ArrayUtils.toObject(intArray), separator)).isEqualTo(expectedIntString);
+
+        assertThat(StringUtils.join(charArray, separatorChar)).isEqualTo(expectedCharString);
+        assertThat(StringUtils.join(ArrayUtils.toObject(charArray), separator)).isEqualTo(expectedCharString);
+
+    }
+
+    @Test
+    public void givenPrimitiveArray_whenJoinBySeparator_thenReturnsString_through_GuavaJoiner() {
+        assertThat(Joiner.on(separator).join(Ints.asList(intArray))).isEqualTo(expectedIntString);
+
+        assertThat(Joiner.on(separator).join(Chars.asList(charArray))).isEqualTo(expectedCharString);
+    }
+
+    @Test
+    public void givenPrimitiveArray_whenJoinBySeparator_thenReturnsString_through_Java7StringBuilder() {
+        assertThat(joinWithStringBuilder(intArray, separator)).isEqualTo(expectedIntString);
```

Comment:
go ahead and show a test method for the char array, even though it is the same

Reply:
Done